### ISSUE TITLE
gitea_repo data resource produces 404 error

### ISSUE THREAD

**Issue by PatDyn** [Lerentis/terraform-provider-gitea#25]
**Describe the bug**
After repo creation:
When calling the gitea_repo data source a 404 Not Found Error is returned.

**To Reproduce**

Create a gitea_repo resource and apply, then create a data source and do ```terraform plan```.

Example Code:
```terraform
data "gitea_repo" "somerepo" {
    depends_on = [
      gitea_repository.somerepo
    ]
    name = module.common.somerepo
    username = "someuser"
}
```

**Expected behavior**
Being able to use the data source to get some basic data about the repo.

**Log Output**
│ Error: 404 Not Found
│ 
│   with data.gitea_repo.somerepo,
│   on gitea_datasources.tf line 4, in data "gitea_repo" "somerepo":
│    4: data "gitea_repo" "somerepo" {

**Additional Data**
Important for reproducability.

- Terraform Version: 1.0.8

- Operating System: kubuntu 22.04

- Provider Version: 0.12.1

- Gitea Version: 1.17.3

**Comment by Lerentis** [comment]
Hi @PatDyn ,

Thanks for the report. I will have a closer look at this probaby over the weekend and check if I can reproduce this

**Comment by Lerentis** [comment]
Fixed with https://github.com/Lerentis/terraform-provider-gitea/commit/2c787bd532f2e6d5cfaae44bb23a52ecf19943e9 and released as [0.12.2](https://github.com/Lerentis/terraform-provider-gitea/releases/tag/v0.12.2)
